src/pkg/services/m365/api/contacts.go

```diff
@@ -117,6 +117,56 @@ func (c Contacts) GetContainerByID(
 	return c.GetFolder(ctx, userID, containerID)
 }
 
+// GetContainerByName fetches a folder by name
+func (c Contacts) GetContainerByName(
+	ctx context.Context,
+	// parentContainerID needed for iface, doesn't apply to contacts
+	userID, _, containerName string,
+) (graph.Container, error) {
+	filter := fmt.Sprintf("displayName eq '%s'", containerName)
+	options := &users.ItemContactFoldersRequestBuilderGetRequestConfiguration{
+		QueryParameters: &users.ItemContactFoldersRequestBuilderGetQueryParameters{
+			Filter: &filter,
+		},
+	}
+
+	ctx = clues.Add(ctx, "container_name", containerName)
+
+	resp, err := c.Stable.
+		Client().
+		Users().
+		ByUserId(userID).
+		ContactFolders().
+		Get(ctx, options)
+	if err != nil {
+		return nil, graph.Stack(ctx, err).WithClues(ctx)
+	}
+
+	gv := resp.GetValue()
+
+	if len(gv) == 0 {
+		return nil, clues.New("container not found").WithClues(ctx)
+	}
+
+	// We only allow the api to match one container with the provided name.
+	// Return an error if multiple container exist (unlikely) or if no container
+	// is found.
+	if len(gv) != 1 {
```

Comment:
are there any (possibly manual) tests we need to do to ensure this won't block restores if the user has contacts folders that we haven't considered at the moment? For example, would something like `contacts/contacts` possible? Would it throw this off?

We can probably punt on folder nesting, but we should add it as a known issues

Reply:
Should be safe.  `filter=displayName eq 'contacts'` doesn't return anything normally.  No reason for it to get confused if there's a folder named 'contacts'.